llvm/lib/Target/AIE/AIELegalizerHelper.cpp

```diff
@@ -142,7 +142,7 @@ bool AIELegalizerHelper::legalizeG_BUILD_VECTOR(LegalizerHelper &Helper,
   assert(DstVecSize < 1024 && "vadd takes a 512-bit argument");
 
   // If our vector is 32-bit we can store it as packed integer vector
-  if (DstVecSize == 32)
+  if (DstVecSize == 32 && MI.getOpcode() == TargetOpcode::G_BUILD_VECTOR)
```

Comment:
Why do we need that extra check? Is the handling of 32-bit vectors different for `G_BUILD_VECTOR_TRUNC`?